From 37ef38092ab783d0126922e8d463024341c481b9 Mon Sep 17 00:00:00 2001
From: Hans Wennborg <hans@chromium.org>
Date: Sat, 4 May 2024 00:19:31 +0000
Subject: [PATCH] Fix base::IDMap::Iterator::operator=

It was referring to non-existing members of Iterator.

Presumably this wasn't noticed because the compiler was doing the name
lookup at instantiation time and this function was never instantiated,
but a recent Clang change [1] uncovered it.

 1. https://github.com/llvm/llvm-project/pull/90152

Bug: 338536260
Change-Id: I16fc766972cabfea117ff6ed7e79f89d0cda475b
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/5512719
Auto-Submit: Hans Wennborg <hans@chromium.org>
Reviewed-by: Daniel Cheng <dcheng@chromium.org>
Commit-Queue: Daniel Cheng <dcheng@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1296463}
---
 base/containers/id_map.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/base/containers/id_map.h b/base/containers/id_map.h
index 4ff5ee7c8bdb35..f84c7c5c6f5e1b 100644
--- a/base/containers/id_map.h
+++ b/base/containers/id_map.h
@@ -178,8 +178,8 @@ class IDMap final {
     }
 
     const Iterator& operator=(const Iterator& iter) {
-      map_ = iter.map;
-      iter_ = iter.iter;
+      map_ = iter.map_;
+      iter_ = iter.iter_;
       Init();
       return *this;
     }
